be/src/column/column_helper.cpp

```diff
@@ -200,6 +200,36 @@ ColumnPtr ColumnHelper::create_const_null_column(size_t chunk_size) {
     return ConstColumn::create(nullable_column, chunk_size);
 }
 
+size_t ColumnHelper::find_nonnull(const Column* col, size_t start, size_t end) {
+    DCHECK_LE(start, end);
+
+    if (!col->has_null()) {
+        return 0;
+    }
+    auto& null = as_raw_column<NullableColumn>(col)->immutable_null_column_data();
+    return SIMD::find_zero(null, start, end - start);
+}
+
+size_t ColumnHelper::last_nonnull(const Column* col, size_t start, size_t end) {
+    DCHECK_LE(start, end);
+    DCHECK_LE(end, col->size());
+
+    if (!col->has_null()) {
+        return end - 1;
+    }
+    auto& null = as_raw_column<NullableColumn>(col)->immutable_null_column_data();
+    for (size_t i = end - 1;;) {
+        if (null[i] == 0) {
+            return i;
+        }
+        if (i == start) {
+            break;
+        }
+        i--;
+    }
```

Comment:
suggest to write like this
```
    for (size_t i = end - 1; i!=start;--i) {
        if (null[i] == 0) {
            return i;
        }
    }
```

Reply:
Well, it's different:
- As `for (size_t = end-1; i != start; --i)`, the `start` is exclusive
- As `if (i == start) break; i--;`, the `start` is inclusive